microcloud/cmd/microcloud/main_init.go

```diff
@@ -82,16 +82,21 @@ func (c *cmdInit) Run(cmd *cobra.Command, args []string) error {
 	}
 
 	services := []service.Service{*cloud, *lxd}
-	_, err = microcluster.App(context.Background(), microcluster.Args{StateDir: api.MicroCephDir})
+	app, err := microcluster.App(context.Background(), microcluster.Args{StateDir: api.MicroCephDir})
 	if err != nil {
-		logger.Info("Skipping MicroCeph service, could not detect state directory")
-	} else {
+		return err
+	}
+
+	_, err = os.Stat(app.FileSystem.ControlSocket().URL.Host)
```

Comment:
@masnax I'm interested/confused how a value containing "URL" and "Host" can be used to stat a directory.

It would make sense if the directory was named after the host, but then I'd expect it to be prefixed with a common path, but in this case it appears to be an absolute path stored in `.Host`. 

Am I missing something here?

Reply:
```go
// ControlSocket returns the full path to the control.socket file that this daemon is listening on.
func (s *OS) ControlSocket() api.URL {
	return *api.NewURL().Scheme("http").Host(filepath.Join(s.StateDir, "control.socket"))
}
```

`LXD Cloud` uses this same method for passing around the control socket. It was mainly just a convenience to allow passing the control socket directly to a client function like a URL.

Reply:
I see, thank you.